### ISSUE TITLE
Could you also include code for ag feature generation

### ISSUE THREAD

**Issue by lithces** [gzBiomedical/EpiScan#2]
I am trying to test this model on some new antigens.

In con_pdb_dict_AgAb.pickle there are three versions of data, of sizes 3, 46, 66 respectively.

Could you add some documents on these features and release code to generate this file?

Btw, the inference instructions in the readme does not work.

**Comment by gzBiomedical** [comment]
I have updated the README.md to provide explanations on how the features are constructed. Additionally, I have tested the inference file and it should now work correctly, addressing your issue. Please check the updated instructions. 
Note that the antibody data used in con_pdb_dict_AgAb.pickle should be of 66 dimensions, while the 3-dimensional and 46-dimensional samples are used for other experiments.

**Comment by lithces** [comment]
Thank you so much for the prompted updates. 

Could you double confirm the index in the readme?

I saw negative (seems integer) value in index 43 (python index), also the last dimension 65 (python index) is not mentioned in the README.

tensor([-1., -3., -4., -3., -2.,  2.,  2., -7.,  1.,  6., -4.,  3.,  3., -7.,
        -6., -5., -4., -4.,  5., -4., -2.,  0., -6., -4., -5.,  2.,  4., -5.,
        -1., -5.,  5.,  4., -4.,  3.,  4., -4., -4., -2., -5.,  4., -5., -4.,
        -4., -3.,  2., -6.,  7., -4.,  6.,  6., -2., -5., -5., -3., -4.,  4.,
        -1., -1., -4.,  0., -3., -2., -4., -1., -2., -1., -4., -2., -3., -3.,
         2.,  0., -4., -1.,  4., -3., -4.,  4., -3., -5.,  0., -5., -3., -4.,
         3., -3., -3., -2., -4., -3.,  3.,  0., -5., -4.,  0.,  2., -1., -3.,
         3., -4., -2., -3.,  1., -3., -5., -3., -3.,  3., -3., -5.,  5.,  0.,
         4.,  5.,  2., -2., -1., -5., -3., -3., -2., -2., -2.,  1., -2., -4.,
        -3.,  5.,  2., -2.,  0.,  3., -2., -3., -1., -5., -4., -4., -5.,  5.,
         1.,  5.,  1., -2.,  6., -6.,  5., -5., -3., -4.,  3., -1., -1., -3.,
        -5.,  1., -2., -2.,  0., -3., -4., -2., -4., -1., -4., -4., -5.,  4.,
        -3.,  2.,  4., -4., -5.,  4., -3., -2., -1., -2., -5.,  0., -2., -3.,
         0.,  0.,  1., -2., -4.,  1., -5., -4.,  1.,  3., -3., -4.,  2., -5.],

What is more, to get the on-hot representation, what is the order of AA for the one-hot?

**Comment by gzBiomedical** [comment]
> Thank you so much for the prompted updates.
> 
> Could you double confirm the index in the readme?
> 
> I saw negative (seems integer) value in index 43 (python index), also the last dimension 65 (python index) is not mentioned in the README.
> 
> tensor([-1., -3., -4., -3., -2., 2., 2., -7., 1., 6., -4., 3., 3., -7., -6., -5., -4., -4., 5., -4., -2., 0., -6., -4., -5., 2., 4., -5., -1., -5., 5., 4., -4., 3., 4., -4., -4., -2., -5., 4., -5., -4., -4., -3., 2., -6., 7., -4., 6., 6., -2., -5., -5., -3., -4., 4., -1., -1., -4., 0., -3., -2., -4., -1., -2., -1., -4., -2., -3., -3., 2., 0., -4., -1., 4., -3., -4., 4., -3., -5., 0., -5., -3., -4., 3., -3., -3., -2., -4., -3., 3., 0., -5., -4., 0., 2., -1., -3., 3., -4., -2., -3., 1., -3., -5., -3., -3., 3., -3., -5., 5., 0., 4., 5., 2., -2., -1., -5., -3., -3., -2., -2., -2., 1., -2., -4., -3., 5., 2., -2., 0., 3., -2., -3., -1., -5., -4., -4., -5., 5., 1., 5., 1., -2., 6., -6., 5., -5., -3., -4., 3., -1., -1., -3., -5., 1., -2., -2., 0., -3., -4., -2., -4., -1., -4., -4., -5., 4., -3., 2., 4., -4., -5., 4., -3., -2., -1., -2., -5., 0., -2., -3., 0., 0., 1., -2., -4., 1., -5., -4., 1., 3., -3., -4., 2., -5.],
> 
> What is more, to get the on-hot representation, what is the order of AA for the one-hot?

I have updated the README file to reflect the correct indices and their meanings. Specifically:
Index 43 (Python index) corresponds to the Conservation Profile (PSSM).
Indices 64 and 65 (Python index) correspond to absolute and relative accessibility, respectively.
Regarding the one-hot encoding, it follows the order inherited from the “Epipred” dataset, which was not utilized in this study. For those interested, you can refer to the original paper via the DOI: 10.1093/bioinformatics/btu190.

**Comment by lithces** [comment]
Thanks for the clarification.

I think the order of AA does matter in this study, at least for "Local Amino Acid Profile (Indices [44-63]). Could you elaborate more on the AA orders?

**Comment by lithces** [comment]
I just check the sum of each row for Local Amino Acid Profile (Indices [44-63]), it is not 1. Could you add more details on how the "frequency" is defined? Are they percentage for each AA type within the neighborhood defined by 8A?

**Comment by gzBiomedical** [comment]
> Local Amino Acid Profile

In the Local Amino Acid Profile function, the 20-dimensional amino acid sequence is arranged according to the alphabetical order of single-letter codes:
{‘A’: 0, ‘C’: 1, ‘D’: 2, ‘E’: 3, ‘F’: 4, ‘G’: 5,
‘H’: 6, ‘I’: 7, ‘K’: 8, ‘L’: 9, ‘M’: 10, ‘N’: 11,
‘P’: 12, ‘Q’: 13, ‘R’: 14, ‘S’: 15, ‘T’: 16, ‘V’: 17,
‘W’: 18, ‘Y’: 19}

Each row in the local_profile matrix is a 20-dimensional vector. For each position in the antigen sequence, it counts the occurrences of corresponding amino acids within the contact range (less than 8 Angstroms). The values are then normalized by column(from sklearn.preprocessing import normalize), which is why you see values between 0-1 (though their sum is not necessarily 1). I’m considering writing a script for data generation (but it might not be ready soon).

**Comment by 21-10-4** [comment]
Indices [0-2] represent the 3D coordinates of the amino acids of the antigenic molecules.
代码：
```
from Bio import PDB
parser = PDB.PDBParser()
structure = parser.get_structure('ID', 'protein.pdb')
coordinates = [atom.get_coord() for atom in structure.get_atoms()]
```
我想请教一下，这个代码得到的是组成残基的 `原子`的坐标，如何得到对应`氨基酸`的3D坐标。

**Comment by gzBiomedical** [comment]
> Indices [0-2] represent the 3D coordinates of the amino acids of the antigenic molecules. 代码：
> 
> ```
> from Bio import PDB
> parser = PDB.PDBParser()
> structure = parser.get_structure('ID', 'protein.pdb')
> coordinates = [atom.get_coord() for atom in structure.get_atoms()]
> ```
> 
> 我想请教一下，这个代码得到的是组成残基的 `原子`的坐标，如何得到对应`氨基酸`的3D坐标。

我是 对于每个残基，获取其所有原子的坐标，并计算这些坐标的平均值，作为该残基的质心坐标。你也可以只取Cα碳原子坐标作为该残基坐标。

**Comment by 21-10-4** [comment]
> > Indices [0-2] represent the 3D coordinates of the amino acids of the antigenic molecules. 代码：
> > ```
> > from Bio import PDB
> > parser = PDB.PDBParser()
> > structure = parser.get_structure('ID', 'protein.pdb')
> > coordinates = [atom.get_coord() for atom in structure.get_atoms()]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 我想请教一下，这个代码得到的是组成残基的 `原子`的坐标，如何得到对应`氨基酸`的3D坐标。
> 
> 我是 对于每个残基，获取其所有原子的坐标，并计算这些坐标的平均值，作为该残基的质心坐标。你也可以只取Cα碳原子坐标作为该残基坐标。

非常感谢🥰
